core/internal/runconfig/runconfig.go

```diff
@@ -165,6 +165,33 @@ func (runConfig *RunConfig) Serialize(format ConfigFormat) ([]byte, error) {
 	return nil, fmt.Errorf("config: unknown format: %v", format)
 }
 
+// Filters the configuration tree based on the given paths.
+//
+// If `exclude` is true, the paths are excluded from the tree. Otherwise, only
+// the paths are included in the tree.
+func (runConfig *RunConfig) FilterTree(
+	paths []RunConfigPath,
+	exclude bool,
+) (RunConfigDict, error) {
+	if exclude {
+		copy := *runConfig
+		for _, path := range paths {
+			copy.removeAtPath(path)
+		}
+		return copy.tree, nil
```

Comment:
This modifies the underlying `tree` even if you copy `runConfig`. It's only a shallow copy, so `removeAtPath()` modifies shared subtrees. You'll need to reconstruct the tree from scratch, making sure to recursively copy subtrees.

I think for this purpose it'll be best to add a new representation of a tree: `type PathMap map[RunConfigPath]interface{}`. This makes the algorithm really simple as long as you have two conversion functions:

```go
pathMap := toPathMap(tree)

for _, path in excludedPaths {
    delete(pathMap, path)
}

return toRunConfigDict(pathMap)
```